include/index_factory.h

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "index.h"
+#include "common_includes.h"
+
+namespace diskann
+{
+template <typename T, typename TagT, typename LabelT> class MemoryIndex;
+template <typename T, typename LabelT> class DiskIndex;
+enum BuildType
+{
+    MEMORY,
+    DISK
+};
+
+enum LoadStoreStratagy
+{
+    GraphStore,
+    DataStore
```

Comment:
what is the enumeration over?

Reply:
This was intended to provide an argument to the index constructor so that we can tie Ravi and Gopl's work to mine. The basic idea was to take this and pass it as an argument so that it can be used to factory instances of Abstract Data Store vs Abstract Graph Store.